telegram/ext/filters.py

```diff
@@ -603,6 +603,45 @@ def filter(self, message: Message) -> bool:
         xml = mime_type('application/xml')
         zip = mime_type('application/zip')
 
+        class file_extension(MessageFilter):
+            """This filter filters documents by their file ending/extension.
+
+            Note:
+                This Filter only filters by the file ending/extension of the document,
+                    it doesn't check the validity of document.
+                The user can manipulate the file extension of a document and
+                    send media with wrong types that don't fit to this handler.
+
+            Example:
+                ``Filters.document.file_extension('jpg')`` filters files with extension ``.jpg``.
+            """
+
+            file_extension: Optional[List[str]] = None
+
+            def __init__(self, file_extension: Optional[str]):
+                """Initialize the extension you want to filter.
+
+                Args:
+                    file_extension (str, optional): media file extension you want to filter.
+                """
+                if file_extension is not None:
+                    file_extension = file_extension.lstrip(".")
+                    self.file_extension = file_extension.split(".")
+                self.name = f"Filters.document.file_extenision({file_extension!r})"
+
+            def filter(self, message: Message) -> bool:
+                """"""  # remove method from docs
+                if message.document is None:
+                    return False
+                if self.file_extension is None:
+                    return True
+                filename_parts = message.document.file_name.split(".")
+                extension_len = len(self.file_extension)
+                if len(filename_parts) - extension_len < 1:
+                    # Exclude extension-only filenames.
+                    return False
+                return filename_parts[-extension_len:] == self.file_extension
```

Comment:
why the hole splitting? `return message.document.file_name.endswith(f'.{self.file_extension}')` should do the trick

Reply:
It mattered when i `lstrip()`ed dots.